### ISSUE TITLE
mark BulkTracking plugin as required, not deactivable

### ISSUE THREAD

**Issue by viktorix** [matomo-org/matomo#14553]
Well this was unexpected. Troubleshooting another problem, I decided to upgrade Matomo to the latest beta. It's not a huge installation, so I used update option inside the dashboard. Everything went smoothly, until I logged back in. Now all sites show "No data has been recorded yet".

At first I thought data was wiped. So I went through the database and it does appear data is in the database, but for some reason Matomo isn't recognizing it anymore. The table prefix is correct in the config file, piwik_

It's only tracking data that's not showing up. Websites are still listed, settings are being saved, etc.

I tried running CLI command to reprocess data, that didn't help.

Few details:
- We only import logs into Matomo. No client-side tracking.
- All system checks are good
- PHP 7.2.14
- MariaDB 10.3.12

This is a list of plugins we've deactivated:
```
Annotations
BulkTracking
Contents
CustomPiwikJs
CustomVariables
DBStats
DevicePlugins
DevicesDetection
Ecommerce
Events
ExampleAPI
ExampleCommand
ExampleLogTables
ExamplePlugin
ExampleReport
ExampleSettingsPlugin
ExampleTracker
ExampleUI
ExampleVisualization
Feedback
GeoIp2
Goals
IntranetMeasurable
Live
MobileAppMeasurable
ProfessionalServices
Provider
Resolution
SegmentEditor
TagManager
UserCountry
UserCountryMap
UserId
Widgetize
```

Here's a screenshot of what I'm seeing in database:
![12155597-f373-4bc0-8a81-b75901242552](https://user-images.githubusercontent.com/1692600/59926590-9b707b00-9408-11e9-9189-88695cd0fd45.png)

Any ideas?

**Comment by tsteur** [comment]
I suggest you look into your PHP error logs, and activate Matomo logging to a file https://matomo.org/faq/troubleshooting/faq_115/

Maybe check if visits get recorded in `piwik_log_visit` to know whether it is a tracking or an archiving issue

**Comment by viktorix** [comment]
@tsteur the original problem wasn't with recording visits, it was showing historical data since 2016 in the dashboard. Logging seems to be working fine. We've tried importing dummy data and it worked fine, showing in the dashboard.

I did make progress. After upgrade, I was seeing "No data has been recorded yet" message and no access to dashboard. I didn't see big button that said hide it. After hiding that screen, I can access my historical data in the dashboard again. However, there's still one problem.

On "All Websites" page, all stats are zero. Here's an example:
![image](https://user-images.githubusercontent.com/1692600/59947762-17d08180-943c-11e9-8628-340f80aedaf5.png)

It seems to be some sort of database problem. I enabled logging, including SQL logging. Every time I reload "All Websites" page, I see a new set of notices that apply to different website IDs. Here's an example:

![image](https://user-images.githubusercontent.com/1692600/59948335-ba3d3480-943d-11e9-83d9-2362244fcbab.png)

**Comment by viktorix** [comment]
"No data has been recorded yet" message is back after an hour. Matomo thinks there's no data for some reason.

**Comment by tsteur** [comment]
It's normal this message is back after an hour. We've fixed an issue related to that message recently so the next rc2 might fix the problem. @mattab maybe could release an RC2 Monday morning first thing so people have more time to test. Can always release RC3, RC4, ...

**Comment by tsteur** [comment]
Make sure though that visits are actually recorded, the "no visits" doesn't sound 100% normal.
Check in piwik_log_visit DB table if there are some recent visits if possible.

**Comment by mattab** [comment]
RC2 has been released, can you try upgrade @viktorix ?

**Comment by mattab** [comment]
> This is a list of plugins we've deactivated:

In theory it shouldn't cause issues to deactivate plugins but it could be creating issues. Did you deactivate these plugins to troubleshoot issues or because you don't need them?

**Comment by viktorix** [comment]
@mattab I deactivated those plugins trying to clean things up, since we only do access logs processing. What I didn't realize is that Bulk Tracking plugin is required for log importing to work. When I deactivated it at the end of May, it stopped importing logs so June had no data.

This caused the original issue I created this thread for. Doing some tests last few days, after upgrading to 3.10.0, Matomo will display "No data has been recorded yet" message if current month's data is empty. It doesn't "see" past data. I don't know if that's by design or a bug.

We imported logs for couple of sites for June, and they do not show "No data has been recorded yet" anymore. Yet, other sites without June's data still show that message.

Tested up to 3.10.0-RC4.

**As a side note**, I would suggest to mention in Bulk Tracking plugin's description that it is required for log importing to work.
